sdk/documenttranslator/ai-document-translator/src/models.ts

```diff
@@ -0,0 +1,219 @@
+export interface BatchSubmissionRequest {
+  /** The input list of documents or folders containing documents */
+  inputs: BatchRequest[];
+}
+
+export interface BatchRequest {
+  /** Source of the input documents */
+  source: SourceInput;
+  /** Location of the destination for the output */
+  targets: TargetInput[];
+  /** Storage type of the input documents source string */
+  storageType?: StorageInputType;
+}
+
+export interface SourceInput {
+  /** Location of the folder / container or single file with your documents */
+  sourceUrl: string;
+  /** */
+  filter?: DocumentFilter;
+  /**
+   * Language code
+   * If none is specified, we will perform auto detect on the document
+   */
+  language?: string;
+  /** Storage Source */
+  storageSource?: StorageSource;
+}
+
+export interface DocumentFilter {
+  /**
+   * A case-sensitive prefix string to filter documents in the source path for translation.
+   * For example, when using a Azure storage blob Uri, use the prefix to restrict sub folders for translation.
+   */
+  prefix?: string;
+  /**
+   * A case-sensitive suffix string to filter documents in the source path for translation.
+   * This is most often use for file extensions
+   */
+  suffix?: string;
+}
+
+export interface TargetInput {
+  /** Location of the folder / container with your documents */
+  targetUrl: string;
+  /** Category / custom system for translation request */
+  category?: string;
+  /** Target Language */
+  language: string;
+  /** List of Glossary */
+  glossaries?: Glossary[];
+  /** Storage Source */
+  storageSource?: StorageSource;
+}
+
+export interface Glossary {
+  /**
+   * Location of the glossary.
+   * We will use the file extension to extract the formatting if the format parameter is not supplied.
+   *
+   * If the translation language pair is not present in the glossary, it will not be applied
+   */
+  glossaryUrl: string;
+  /** Format */
+  format: string;
+  /** Optional Version.  If not specified, default is used. */
+  version?: string;
+  /** Storage Source */
+  storageSource?: StorageSource;
+}
+
+export interface ErrorResponseV2 {
+  /** This contains an outer error with error code, message, details, target and an inner error with more descriptive details. */
+  error?: ErrorV2;
+}
+
+export interface ErrorV2 {
```

Comment:
the V2 suffixes confuse me - is it to disambiguate or are they improved versions of the non-V2 types?

Reply:
Yeah it is an ugly name, unfortunately this comes from the Swagger

Reply:
if it's in the swagger I have no issues with the codegen :)